ppod.py

```diff
@@ -25,12 +26,34 @@ def lambda_handler(event: dict, context: object) -> dict:
     )
     for s3_file in s3_files:
         logger.info("Processing file: %s", s3_file)
-        s3_file_content = smart_open.open(f"s3://{bucket}/{s3_file}", "rb")
-        files = extract_files_from_tar(s3_file_content)
-        for file in files:
-            file  # do a thing
-            file_count += 1
-    return {"files-processed": file_count}
+        with smart_open.open(f"s3://{bucket}/{s3_file}", "rb") as s3_file_content:
+            xml_files = extract_files_from_tar(s3_file_content)
+            for xml_file in xml_files:
+                if xml_file:
+                    add_namespaces_to_alma_marcxml(xml_file)
+                    # post modified_xml to POD
+                    file_count += 1
+                else:
+                    raise ValueError(f"No files extracted from {s3_file}")
+    return {"files_processed": file_count}
+
+
+def add_namespaces_to_alma_marcxml(xml_file: IO[bytes]) -> StringIO:
+    collection_element_with_namespaces = (
+        '<?xml version="1.0" encoding="UTF-8"?>\n'
+        '<collection xmlns="http://www.loc.gov/MARC21/slim" '
+        'xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" '
+        'xsi:schemaLocation="http://www.loc.gov/MARC21/slim '
+        'http://www.loc.gov/standards/marcxml/schema/MARC21slim.xsd"><'
+    )
+    xml_string = StringIO()
+    xml_line = xml_file.read(52)
+    if xml_line == b'<?xml version="1.0" encoding="UTF-8"?>\n<collection><':
+        xml_string.write(collection_element_with_namespaces)
+        xml_file.seek(52)
+        xml_string.write(xml_file.read().decode("utf-8"))
```

Comment:
This is still going to cause the same memory issue because `xml_file.read()` is still reading the entire xml file into memory. We want to do something like this:
```
    output = io.BytesIO()
    first_chunk = xml_file.read(51)
    decoded = first_chunk.decode("utf-8")
    if decoded != '<?xml version="1.0" encoding="UTF-8"?>\n<collection>':
        raise ValueError("XML file does not have expected XML declaration")
    output.write(collection_element_with_namespaces.encode())
    while True:
        chunk = xml_file.read(16384)
        if not chunk:
            break
        output.write(chunk)
    output.seek(0)
    return output
```

Assuming POD will accept it, I think we should return a `BytesIO` object here instead of string, both because of the decoding multibyte character issue when reading in chunks and because `requests` suggests sending bytes when streaming large files.

Reply:
I can test whether POD will accept bytes tomorrow. I was assuming the `io.BufferedReader's` `read` method would handle the memory issues but I'm fine doing it this way

Reply:
Sadly it does not. If it did, we could have used the code as it was in the previous commit :)

Reply:
Confirmed that POD will accept a `BytesIO` object with a test of my next PR's code on one of the test streams